src/Orleans.Runtime/Placement/Filtering/PreferredMatchSiloMetadataPlacementFilterStrategy.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using Orleans.Metadata;
+using Orleans.Placement;
+
+namespace Orleans.Runtime.Placement.Filtering;
+
+public class PreferredMatchSiloMetadataPlacementFilterStrategy(string[] orderedMetadataKeys, int minCandidates, int order)
+    : PlacementFilterStrategy(order)
+{
+    public string[] OrderedMetadataKeys { get; set; } = orderedMetadataKeys;
+    public int MinCandidates { get; set; } = minCandidates;
+
+    public PreferredMatchSiloMetadataPlacementFilterStrategy() : this([], 1, 0)
```

Comment:
Should the `minCandidates` parameter here also default to `2`, to match what's given by `PreferredMatchSiloMetadataPlacementFilterAttribute` ?
```suggestion
    public PreferredMatchSiloMetadataPlacementFilterStrategy() : this([], 2, 0)
```

Reply:
No reason it couldn't, though it doesn't actually make any difference since the only time the parameterless constructor is called is in a code path that then immediately calls `Initialize` which sets the internal properties based on the GrainProperties dictionary for the Grain Type being activated. So, these values never end up getting used for anything.